custom_components/google_home/config_flow.py

```diff
@@ -54,6 +54,11 @@ async def async_step_user(
             username = user_input[CONF_USERNAME]
             password = user_input[CONF_PASSWORD]
             session = async_create_clientsession(self.hass)
+            try:
+                master_token = user_input[CONF_MASTER_TOKEN]
+            except KeyError:
+                # no master token provided
+                master_token = None
```

Comment:
Bummer, my comment did get submitted while the first review :/ 

Would have been nice to add input validation here.

Something inline with:
- make all 3 values `username`, `password` and `master_token` optional inputs.
- `.get` all 3 values, fallback to `None` 
- validate that either:
  - username and password are provided
  or 
  - master_token is provided
- instantiate client

Reply:
Sounds good. I've updated the PR regarding this suggestion. 
Still, if an invalid `master_token` is provided, the integration will just finish the set-up process but with no devices configured. This may be a thing which could also be improved. My idea was to check if it possible to retrieve an `access_token`, but not sure if this should be included here since this is part of glocaltokens

Reply:
yes, that would be great. 
The right place for token validation and credential validation is probably in `glocaltokens`, you are right. 
But as of now, if you check `_test_credentials` method in this package it calls for `/api/GlocaltokensApiClient/async_get_master_token` that calls to get a master token using credentials to validate if credentials worked out. 
We can just expand that functionality and add `GlocaltokensApiClient.async_get_access_tokens` in the same way, which will call underlying `glocaltokens` client to load access tokens. What do you think?

Reply:
Was my initial thought, too. Implemented it this way and already pushed it to the PR :)